### PR TITLE
feat: support no ledger

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What?
- Implemented UI for selecting ledger and non-ledger based methods, networks, and inputs.
- Refactored functionality for spinning up a wallet.
- Refactored mapping logic for selected ledger, method, and network.
- Refactored UI for the "Create DID" popup.
- Applied restrictions for the "Create DID" button for the issuer role.
- Refactored the "Create DID" functionality.

### CODE REVIEW

File: src/components/organization/configuration-settings/CreateDid.tsx [hunk 1]
```diff
@@ -1,121 +1,147 @@
...
+
+			const provider = new ethers.JsonRpcProvider(testnetUrl)
+
+			const wallet = new ethers.Wallet(privateKey, provider);
+			const address = await wallet.getAddress();
+			const balance = await provider.getBalance(address);
+
+			const etherBalance = ethers.formatEther(balance);
+
+			if (parseFloat(etherBalance) < 0.01) {
```
Comment (tipusinghaw) [inline src/components/organization/configuration-settings/CreateDid.tsx:104]: Add the comparison data in enum or in a global variable

File: src/components/organization/configuration-settings/CreateDid.tsx [hunk 2]
```diff
@@ -1,121 +1,147 @@
...
 	};
 
 	useEffect(() => {
 		fetchOrganizationDetails();
 	}, []);
 
+
+	const checkBalance = async (privateKey: string) => {
+		try {
+			const testnetUrl = 'https://rpc-amoy.polygon.technology';
```
Comment (tipusinghaw) [inline src/components/organization/configuration-settings/CreateDid.tsx:94]: Please add RPC URL in .env

File: src/components/organization/configuration-settings/CreateDid.tsx [hunk 3]
```diff
@@ -1,121 +1,147 @@
...
 		}
-		setLoading(false);
 	};
 
 	useEffect(() => {
 		fetchOrganizationDetails();
 	}, []);
 
+
+	const checkBalance = async (privateKey: string) => {
```
Comment (tipusinghaw) [inline src/components/organization/configuration-settings/CreateDid.tsx:92]: This function should also handle the mainnet flow

File: src/components/organization/configuration-settings/CreateDid.tsx [hunk 4]
```diff
@@ -250,137 +242,62 @@ const CreateDIDModal = (props: EditOrgdetailsModalProps) => {
...
-
-						if (ledgerName !== selectedLedgerName) {
-							setErrMsg('This ledger is not applicable to create a DID');
-						} else {
-							await createNewDid(values);
-							setErrMsg(null);
-							window.location.reload();
-						}
+						await createNewDid(values);
+						window.location.reload();
```
Comment (tipusinghaw) [inline src/components/organization/configuration-settings/CreateDid.tsx:247]: Any specific reason for using reload() method

File: src/components/organization/walletCommonComponents/SharedAgent.tsx [hunk 5]
```diff
@@ -22,561 +19,457 @@ const SharedAgentForm = ({
...
-					<span>{mappedData[selectedLedger][selectedLedger] || ''}</span>
-				) : (
-					<></>
-				);
-			}
-			default:
-				return <></>;
-		}
-	};
+		setSeedVal(seeds);
```
Comment (tipusinghaw) [inline src/components/organization/walletCommonComponents/SharedAgent.tsx:110]: Please discuss the this flow

File: src/components/organization/walletCommonComponents/SharedAgent.tsx [hunk 6]
```diff
@@ -22,561 +19,457 @@ const SharedAgentForm = ({
...
+														to get free token
+													</div>
+												</div>
+											</li>
+											<li className='mb-2'>
+												<span className='font-semibold text-sm'>Step 2:</span>
+												<div className='ml-4 text-sm text-gray-700'>Check that you have recieved the tokens.</div>
+												<div className='ml-4 text-sm text-gray-700'>For eg. copy the address and check the balance on
+													<div>
+														<a href='https://mumbai.polygonscan.com/' className='text-blue-900 text-sm underline'>
```
Comment (tipusinghaw) [inline src/components/organization/walletCommonComponents/SharedAgent.tsx:448]: URL should be in .env

**Review by pranalidhanavade — APPROVED** [review]
LG

File: src/components/organization/walletCommonComponents/SharedAgent.tsx (hunk 5 above)
Comment (bhavanakarwade) [reply src/components/organization/walletCommonComponents/SharedAgent.tsx:110]: Resolved

File: src/components/organization/configuration-settings/CreateDid.tsx (hunk 4 above)
Comment (bhavanakarwade) [reply src/components/organization/configuration-settings/CreateDid.tsx:247]: Resolved

File: src/components/organization/configuration-settings/CreateDid.tsx (hunk 3 above)
Comment (bhavanakarwade) [reply src/components/organization/configuration-settings/CreateDid.tsx:92]: Resolved

File: src/components/organization/walletCommonComponents/SharedAgent.tsx (hunk 6 above)
Comment (bhavanakarwade) [reply src/components/organization/walletCommonComponents/SharedAgent.tsx:448]: Resolved

File: src/components/organization/configuration-settings/CreateDid.tsx (hunk 2 above)
Comment (bhavanakarwade) [reply src/components/organization/configuration-settings/CreateDid.tsx:94]: Resolved

File: src/components/organization/configuration-settings/CreateDid.tsx (hunk 1 above)
Comment (bhavanakarwade) [reply src/components/organization/configuration-settings/CreateDid.tsx:104]: Resolved

**Review by tipusinghaw — APPROVED** [review]
LGTM

**Pull request merged**